docs/architecture/deployment.md

```diff
@@ -0,0 +1,6 @@
+TODO:
```

Comment:
See other comment, this is a lot of stuff to change in 1 PR.

I know this is still WIP, but I wouldn't even put the TODO here and break this up into more modular changes and address this in a separate PR.

I am also working on a PR to explain how Safe accounts are created and was going to put that in safe-docs. We should sync to avoid duplicate work.

Reply:
Yeah, I understand that, I still think it can go in the normal docs.

From my perspective, developer docs should include how to use a tool and an explanation for the design decisions made when creating this tool because developers want to know both.

If we think of the developer's journey, first they might want to know how to use the contracts. Then they will want to know how the contracts work behind the scenes and why certain decisions are made when creating these cotnracts. Understanding the why can also help them in debugging problems and getting ideas for other ways they can use the contracts.

Another way to think about this: there is a big overlap in the audience between people who want to know how to use the contracts and people who want to understand the how and why behind the contracts' creation.

It makes sense to offer them all this information in one place instead of spreading it in different places

This is also guided by feedback from safe developers saying they don't like having to dig through multiple different places to get information.

![Screen Shot 2023-03-24 at 9 56 04 AM](https://user-images.githubusercontent.com/9806858/227540585-e6172661-d118-489d-b337-63830951ff05.png)

Reply:
As this is meant as code documentation, it will be part of the code, which is this repository. Other documentation building on top of it should probably not be in this repository.

Reply:
Okay, that's fine with me :thumbsup: